src/main/java/de/tum/in/www1/artemis/service/ModelingAssessmentService.java

```diff
@@ -120,24 +110,14 @@ public Result saveManualAssessment(Long resultId, Long exerciseId, String modeli
         // write assessment to file system
         jsonAssessmentRepository.writeAssessment(exerciseId, studentId, submissionId, true, modelingAssessment);
         resultRepository.save(result);
-        return result;
     }
 
 
-    /**
-     * Helper function to calculate the total score of an assessment json. It loops through all assessed model elements
-     * and sums the credits up.
-     *
-     * @param assessmentJson    the assessments as JsonObject
-     * @return the total score
-     */
-    private Double calculateTotalScore(JsonObject assessmentJson) {
-        Double totalScore = 0.0;
-        JsonArray assessments = assessmentJson.get("assessments").getAsJsonArray();
-        for (JsonElement assessment : assessments) {
-            totalScore += assessment.getAsJsonObject().getAsJsonPrimitive("credits").getAsDouble();
+    public static Double calculateTotalScore(List<ModelElementAssessment> modelingAssessment) {
+        double totalScore = 0.0;
+        for (ModelElementAssessment assessment : modelingAssessment) {
+            totalScore += assessment.getCredits();
         }
-        //TODO round this value to max two numbers after the comma
-        return totalScore;
+        return new BigDecimal(totalScore).setScale(2, ROUND_HALF_EVEN).doubleValue();
```

Comment:
please add a short comment, what you are achieve with this line. Not everyone is familiar with rounding using BigDecimal

Reply:
I added  a documentation